distribution/lib/Standard/DuckDB/0.0.0-dev/src/DuckDB_Connection.enso

```diff
@@ -821,3 +826,56 @@ private _handle_existing_file path on_existing_file allow_append:Boolean=False =
                         tmp_file.move_to (path.parent / (path.name+".bak")) replace_existing=True
 
             result
+
+private _additional_files extension = case extension.to_case ..Lower of
+    ".tab" ->
+        ## Map Info - 2 files needed .tab, .dat and optionally .id, .shp
+        [".dat", "*.id", "*.shp"]
+    ".shp" ->
+        ## Shapefile - 3 files needed .shp, .shx, .dbf and optionally .prj
+        [".shx", ".dbf", "*.prj"]
+    _ -> []
+
+private _with_additional_files resolved_file action =
+    extras = case resolved_file of
+        _ : File -> []
+        _ : URI ->
+            possible_extension = resolved_file.path.drop (..Before_Last ".")
+            extras = _additional_files possible_extension
+            if extras==[] then Data_Link_Helpers.as_file resolved_file action else
+                Error.throw (Illegal_Argument.Error "Cannot use a URI wit a file type that requires additional files. Please download all related files first.")
+        _ -> _additional_files resolved_file.extension
+
+    if extras==[] then Data_Link_Helpers.as_file resolved_file action else
+        file = Data_Link_Helpers.as_file resolved_file f->f delete_on_return=False
+        Error.return_if_error file
+
+        parent_source = resolved_file.parent
+        base_name = resolved_file.name.drop (..After_Last ".") . drop (..Last 1)
+        parent_dest = file.parent
+        parent_name = file.name.drop (..After_Last ".") . drop (..Last 1)
+
+        ## Check if all required files exist as then can fail fast
+        check_required = extras.map on_problems=No_Wrap.Value extra->
+            required = extra.starts_with "*" . not
+            if required.not then Nothing else
+                extra_name = if required then base_name + extra else base_name + extra.drop (..First 1)
```

Comment:
Don't think you need this if statement given we are inside the if required.not statement already